packages/language-server/lib/convert.js

```diff
@@ -0,0 +1,251 @@
+ * @typedef {import('typescript').DiagnosticCategory} DiagnosticCategory
+ * @typedef {import('typescript').DiagnosticMessageChain} DiagnosticMessageChain
+ * @typedef {import('typescript').DiagnosticRelatedInformation} DiagnosticRelatedInformation
+ * @typedef {import('typescript').JSDocTagInfo} JSDocTagInfo
+ * @typedef {import('typescript').QuickInfo} QuickInfo
+ * @typedef {import('typescript').SymbolDisplayPart} SymbolDisplayPart
+ * @typedef {import('typescript').ScriptElementKind} ScriptElementKind
+ * @typedef {import('typescript').TextSpan} TextSpan
+ * @typedef {import('vscode-languageserver').Diagnostic} LspDiagnostic
+ * @typedef {import('vscode-languageserver').DiagnosticRelatedInformation} LspDiagnosticRelatedInformation
+ * @typedef {import('vscode-languageserver-textdocument').TextDocument} TextDocument
+ */
+
+import {
+  CompletionItemKind,
+  DiagnosticSeverity,
+  DiagnosticTag,
+  LocationLink,
+  Range,
+} from 'vscode-languageserver'
+
+import { documents } from './documents.js'
+
+/**
+ * @param {ScriptElementKind} kind
+ * @returns {CompletionItemKind} The matching Monaco completion item kind.
+ */
+export function convertScriptElementKind(kind) {
+  switch (kind) {
+    case 'primitive type':
+    case 'keyword':
+      return CompletionItemKind.Keyword
+    case 'var':
+    case 'local var':
+      return CompletionItemKind.Variable
+    case 'property':
+    case 'getter':
+    case 'setter':
+      return CompletionItemKind.Field
+    case 'function':
+    case 'method':
+    case 'construct':
+    case 'call':
+    case 'index':
+      return CompletionItemKind.Function
+    case 'enum':
+      return CompletionItemKind.Enum
+    case 'module':
+      return CompletionItemKind.Module
+    case 'class':
+      return CompletionItemKind.Class
+    case 'interface':
+      return CompletionItemKind.Interface
+    case 'warning':
+      return CompletionItemKind.File
+  }
+  return CompletionItemKind.Property
+}
+
+/**
+ * @param {ts} ts
+ * @param {CompletionEntryDetails | QuickInfo} details
+ * @returns {string} XXX
+ */
+export function createDocumentationString(ts, details) {
+  let documentationString = ts.displayPartsToString(details.documentation)
+  if (details.tags) {
+    for (const tag of details.tags) {
+      documentationString += `\n\n${tagToString(tag)}`
+    }
+  }
+  return documentationString
+}
+
+/**
+ * @param {JSDocTagInfo} tag
+ * @returns {string} XXX
+ */
+function tagToString(tag) {
+  let tagLabel = `*@${tag.name}*`
+  if (tag.name === 'param' && tag.text) {
+    const [paramName, ...rest] = tag.text
+    tagLabel += `\`${paramName.text}\``
+    if (rest.length > 0) tagLabel += ` — ${rest.map(r => r.text).join(' ')}`
+  } else if (Array.isArray(tag.text)) {
+    tagLabel += ` — ${tag.text.map(r => r.text).join(' ')}`
+  } else if (tag.text) {
+    tagLabel += ` — ${tag.text}`
+  }
+  return tagLabel
+}
+
+/**
+ * @param {TextDocument} doc
+ * @param {TextSpan} span
+ * @returns {Range} XXX
```

Comment:
Is this a placeholder for future documentation? Or is there an issue with the return?